src/python/WMCore/WMSpec/Steps/Fetchers/PileupFetcher.py

```diff
@@ -70,15 +74,21 @@ def _queryDbsAndGetPileupConfig(self, stepHelper, dbsReader, fakeSites):
             # each dataset input can generally be a list, iterate over dataset names
             blockDict = {}
             for dataset in datasets:
-                blockNames = dbsReader.listFileBlocks(dataset)
-                # DBS listBlocks returns list of DbsFileBlock objects for each dataset,
-                # iterate over and query each block to get list of files
-                for dbsBlockName in blockNames:
-                    blockDict[dbsBlockName] = {"FileList": sorted(dbsReader.lfnsInBlock(dbsBlockName)),
-                                               "PhEDExNodeNames": dbsReader.listFileBlockLocation(dbsBlockName),
-                                               "NumberOfEvents": dbsReader.getDBSSummaryInfo(block=dbsBlockName)['NumberOfEvents']}
-                    blockDict[dbsBlockName]['PhEDExNodeNames'].extend(x for x in fakeSE if x not in \
-                                                                      blockDict[dbsBlockName]['PhEDExNodeNames'])
+            
+                blockFileInfo = dbsReader.dbs.listFileArray(dataset=dataset, validFileOnly = 1, detail=True)
```

Comment:
:-) Actually it might be better to wrap up listFileArray within DBSReader api for maintenance .  I will see whether that make more sense.